### PR TITLE
Overhaul docs to mention only Hip

### PR DESCRIPTION
**Pull request by sethmlarson**
- Remove `CONTRIBUTORS.txt`, I dislike these files and would rather use GitHub contributors. A lot less noise! :)
- Upgrade all `pip installs` to `python -m pip install` per recommendations of PyPA.
- Change all code examples to use `hip.` instead of `urllib3.` 
- Change capitalization on sub-sections.

I got rid of the `contrib` section for now since we're removing pyOpenSSL and SOCKS support doesn't work yet. I figure we'll probably provide a more "integrated" solution?

I also would like to move towards using a Sphinx theme like Trio's, maybe even "mostly" copy? Alabaster doesn't quite capture the "this is a high-level client" feel. :) What are our thoughts here?

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
    a *Contributor Friendly* tag for issues that should be ideal for people who
    are not very familiar with the codebase yet.
-#. Fork the `urllib3 repository on Github <https://github.com/urllib3/urllib3>`_
+#. Fork the `Hip repository on Github <https://github.com/python-trio/hip>`_
    to start making your changes.
 #. Write a test which shows that the bug was fixed or that the feature works
    as expected.
 #. Format your changes with black using command `$ nox -s blacken` and lint your
    changes using command `nox -s lint`.
-#. Send a pull request and bug the maintainer until it gets merged and published.
```
Comment (pquentin) [inline docs/contributing.rst:18]: Let's keep this line :)

**Comment by pquentin** [comment]
Thanks! The docs definitely needed work

File: docs/contributing.rst (hunk 1 above)
Comment (pquentin) [reply docs/contributing.rst:18]: I tried pushing this myself but maybe you didn't check "Allow edit from maintainers"? https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by pquentin** [comment]
> I also would like to move towards using a Sphinx theme like Trio's, maybe even "mostly" copy? Alabaster doesn't quite capture the "this is a high-level client" feel. :) What are our thoughts here?

The cool kids use material themes these days, eg. https://bashtage.github.io/sphinx-material/ for Sphinx. The Trio theme sounds good too.

File: docs/contributing.rst (hunk 1 above)
Comment (sethmlarson) [reply docs/contributing.rst:18]: Done and done!

**Comment by RatanShreshtha** [comment]
material themes looks nice 👍

**Pull request merged**